services/slam/builtin/orbslam_int_test.go

```diff
@@ -66,35 +68,51 @@ func releaseImages(t *testing.T, mode slam.Mode) {
-	t.Logf("Position point: (%v, %v, %v)",
-		position.Pose().Point().X, position.Pose().Point().Y, position.Pose().Point().Z)
-	// Typical values for RGBD are around (0.602, -0.772, -0.202), theta=0.002
-	// Typical values for Mono without an existing map are around (0.144, 0.980, -0.137), theta=0.104
-	// Typical values for Mono with an existing map are around ( 0.092, 0.993, -0.068), theta=0.099
-	t.Logf("Position orientation: RX: %v, RY: %v, RZ: %v, Theta: %v",
-		position.Pose().Orientation().AxisAngles().RX,
-		position.Pose().Orientation().AxisAngles().RY,
-		position.Pose().Orientation().AxisAngles().RZ,
-		position.Pose().Orientation().AxisAngles().Theta)
+// Checks the orbslam map and confirms there are more than zero map points.
+func testOrbslamMap(t *testing.T, svc slam.Service) {
 	actualMIME, _, pointcloud, err := svc.GetMap(context.Background(), "test", "pointcloud/pcd", nil, false, map[string]interface{}{})
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, actualMIME, test.ShouldResemble, "pointcloud/pcd")
-	// Typical value for RGBD is 329
-	// Values for Mono vary
 	t.Logf("Pointcloud points: %v", pointcloud.Size())
 	test.That(t, pointcloud.Size(), test.ShouldBeGreaterThan, 0)
 }
 
+// Checks the orbslam position within a defined tolerance
+func testOrbslamPosition(t *testing.T, svc slam.Service, mode, actionMode string) {
+	var expectedPos r3.Vector
+	expectedOri := &spatialmath.OrientationVector{}
+	tolerancePos := 0.05
+	toleranceOri := 0.5
+
+	switch {
+	case mode == "mono" && actionMode == "mapping":
+		expectedPos = r3.Vector{X: 0.020, Y: -0.032, Z: -0.053}
+		expectedOri = &spatialmath.OrientationVector{Theta: 0.104, OX: 0.144, OY: 0.980, OZ: -0.137}
+	case mode == "mono" && actionMode == "updating":
+		expectedPos = r3.Vector{X: 0.023, Y: -0.036, Z: -0.040}
+		expectedOri = &spatialmath.OrientationVector{Theta: 0.099, OX: 0.092, OY: 0.993, OZ: -0.068}
+	case mode == "rgbd":
+		expectedPos = r3.Vector{X: -0.001, Y: -0.004, Z: -0.008}
+		expectedOri = &spatialmath.OrientationVector{Theta: 0.002, OX: 0.602, OY: -0.772, OZ: -0.202}
+	}
+
+	position, err := svc.Position(context.Background(), "test", map[string]interface{}{})
+	test.That(t, err, test.ShouldBeNil)
+
+	actualPos := position.Pose().Point()
+	t.Logf("Position point: (%v, %v, %v)", actualPos.X, actualPos.Y, actualPos.Z)
+	test.That(t, actualPos.X, test.ShouldBeBetween, expectedPos.X-tolerancePos, expectedPos.X+tolerancePos)
+	test.That(t, actualPos.Y, test.ShouldBeBetween, expectedPos.Y-tolerancePos, expectedPos.Y+tolerancePos)
+	test.That(t, actualPos.Z, test.ShouldBeBetween, expectedPos.Z-tolerancePos, expectedPos.Z+tolerancePos)
+
+	actualOri := position.Pose().Orientation().AxisAngles()
+	t.Logf("Position orientation: RX: %v, RY: %v, RZ: %v, Theta: %v", actualOri.RX, actualOri.RY, actualOri.RZ, actualOri.Theta)
+	test.That(t, actualOri.RX, test.ShouldBeBetween, expectedOri.OX-toleranceOri, expectedOri.OX+toleranceOri)
```

Comment:
Same here, it seems incorrect that we're comparing axis angle representation to orientation vector representation.

Reply:
See previous comment